spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadOptions.java

```diff
@@ -35,6 +35,12 @@ private SparkReadOptions() {}
   // A timestamp in milliseconds; the snapshot used will be the snapshot current at this time.
   public static final String AS_OF_TIMESTAMP = "as-of-timestamp";
 
+  // branch ref of the table snapshot to read from
+  public static final String BRANCH = "branch";
+
+  // tag ref of the table snapshot to read from
```

Comment:
Just a nit: I think for the comments we can leave off the "table snapshot" and "ref" part and the comment could look something like 

"Tag to read from"
"Branch to read from"